codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AwsEndpointAuthSchemeGenerator.java

```diff
@@ -208,17 +208,17 @@ private GoWriter.Writable generateSigV4AResolution(ServiceShape serviceShape) {
             writer.write(
                 """
                     v4aScheme, _ := authScheme.($P)
-                    var signingName string
                     if v4aScheme.SigningName == nil {
-                        signingName = \"$L\"
+                        signingNameDefault := \"$L\"
+                        v4aScheme.SigningName = &signingNameDefault
```

Comment:
you can use this utility instead of having to create an extra variable
`SymbolUtils.createValueSymbolBuilder("String", AwsGoDependency.AWS_CORE).build()`